ibis/expr/rules.py

```diff
@@ -425,9 +425,12 @@ def table(arg, *, schema=None, **kwargs):
 @validator
 def column_from(name, column, *, this):
     if not isinstance(column, (str, int)):
-        raise com.IbisTypeError(
-            f"value must be an int or str, got {type(column).__name__}"
-        )
+        if isinstance(column, ir.AnyColumn):
+            column = column.op().name
```

Comment:
You can't just look at the name, because a column operation's name (or index) may match `column` while originating from a different table.

Reply:
You can do something like:

```python
if isinstance(column, ir.ColumnExpr):
    # return column if getattr(this, name)[column_name] equals column, otherwise fail
else:
    # existing logic
```